construct/lib/container.py

```diff
@@ -77,27 +77,45 @@ def __init__(self, *args, **kw):
             self[k] = v
 
     def __getstate__(self):
-        return list(self.items())
+        return self.__keys_order__
 
     def __setstate__(self, state):
-        object.__setattr__(self, "__keys_order__", [])
-        dict.clear(self)
-        self.update(state)
+        self.__keys_order__ = state
 
     def __getattr__(self, name):
         try:
-            return self[name]
+            if name in self.__slots__:
+                try:
+                    return object.__getattribute__(self, name)
+                except AttributeError as e:
+                    if name == "__keys_order__":
+                        object.__setattr__(self, "__keys_order__", [])
+                        return []
+                    else:
+                        raise e
+            else:
+                return self[name]
         except KeyError:
             raise AttributeError(name)
 
     def __setitem__(self, key, val):
-        if key not in self:
-            self.__keys_order__.append(key)
-        dict.__setitem__(self, key, val)
+        if key in self.__slots__:
+            object.__setattr__(self, key, val)
+        else:
```

Comment:
I'll try to do this tomorrow.

However, this is already the case in the current master...